tests/test_numbers.py

```diff
@@ -154,6 +154,8 @@ def test_compact(self):
         assert numbers.format_compact_decimal(-123456789, format_type='long', locale='en_US') == u'-123 million'
         assert numbers.format_compact_decimal(2345678, locale='mk', format_type='long') == u'2 милиони'
         assert numbers.format_compact_decimal(21098765, locale='mk', format_type='long') == u'21 милион'
+        assert numbers.format_compact_decimal(decimal.Decimal(123456789), format_type='short', locale='en_US') == u'123M'
+        assert numbers.format_compact_decimal('123456789', format_type='short', locale='en_US') == u'123M'
```

Comment:
I don't think the test passing `str` is needed.

The `format_compact_decimal` uses `NumberPattern.apply` as the same of `format_decimal`. docstring of the `apply` says `:type value: decimal.Decimal|float|int`. `str` instance is not expected.

Actually, `test_format_decimal` doesn't have tests passing `str` instance to `format_decimal`

Reply:
`str` works differently in `format_compact_decimal` than in `format_decimal` since it gets passed through `_get_compact_format` which previously did not support string because of the `abs()` call